scipy/optimize/_trustregion_constr/minimize_trustregion_constr.py

```diff
@@ -18,7 +18,8 @@
     0: "The maximum number of function evaluations is exceeded.",
     1: "`gtol` termination condition is satisfied.",
     2: "`xtol` termination condition is satisfied.",
-    3: "`callback` function requested termination."
+    3: "`callback` function requested termination.",
+    4: "Constraints are not satisfied."
```

Comment:
I think the status message can be improved here, currently it doesn't give enough information as to why the status was set to 4.

How about `Constraint violation exceeds 'gtol'`

Reply:
I think you are right.  Your status message is more clear.  I'll go ahead and update it.

Reply:
Quick question.  Should I rebase before attempting another commit? This is my first PR and am not familiar with the process to update an existing commit.

Reply:
Feel free to just commit the change on top of your current branch - a reviewer can always 'squash' commits down before merge.

Reply:
Thanks, I just added a new commit with the updated status message.